purchase_order_generator/models/purchase_order_generator.py

```diff
@@ -84,8 +84,17 @@ def generate_cpo(self):
                 [
                     ("product_tmpl_id", "=", product_id),
                     ("name", "=", supplier.id),
-                ]
+                ],
+                order="date_start desc",
+                limit=1,
             )
+
+            if not supplierinfo:
+                product_name = self.env["product.template"].browse(product_id)
```

Comment:
browsing in a for loop is bad for perf issue. you should loop on the browsed products

Doing something like for product in self.env["product.template"].browse(product_ids):